test: replace ganache with anvil

mattsse opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests.

Contributors guide: https://github.com/gakonst/ethers-rs/blob/master/CONTRIBUTING.md

The contributors guide includes instructions for running rustfmt and building the
documentation.
-->

## Motivation
Closes https://github.com/gakonst/ethers-rs/issues/1227

supersedes #1229
<!--
Explain the context and why you're making that change. What is the problem
you're trying to solve? In some cases there is not a problem and this can be
thought of as being the motivation for your change.
-->

## Solution

<!--
Summarize the solution and provide any necessary context needed to understand
the code change.
-->

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

gakonst commented:
I also think we may want to replace all `Geth` instances with `Anvil` in the tests? Found a mempool API which we do not implement in Anvil:

```
running 1 test
test txpool ... FAILED

failures:

---- txpool stdout ----
thread 'txpool' panicked at 'called `Result::unwrap()` on an `Err` value: JsonRpcClientError(JsonRpcError(JsonRpcError { code: -32601, message: "Method not found", data: None }))', ethers-providers/tests/txpool.rs:28:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

for the call `provider.txpool_status()`

gakonst commented:
I also think we may want to replace all `Geth` instances with `Anvil` in the tests? Found a mempool API which we do not implement in Anvil:

```
running 1 test
test txpool ... FAILED

failures:

---- txpool stdout ----
thread 'txpool' panicked at 'called `Result::unwrap()` on an `Err` value: JsonRpcClientError(JsonRpcError(JsonRpcError { code: -32601, message: "Method not found", data: None }))', ethers-providers/tests/txpool.rs:28:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

for the call `provider.txpool_status()`, supect it may apply to the `txpool_inspect()` too

gakonst commented:
@mattsse the commit above highlights some new issue in the mempool implementation I think

mattsse commented:
> @mattsse the commit above where I replaced Geth txpool with Anvil highlights some new issue in the mempool implementation I think

used the current gas price rather than the gas price set by the tx, fixed here https://github.com/foundry-rs/foundry/pull/1756 which should unblock this once included in nightly
